### PR TITLE
[COST-5134] Update OpenShift tag matching to be key value pairs

### PR DESCRIPTION
**Pull request by myersCody**
## Jira Ticket

[COST-5134](https://issues.redhat.com/browse/COST-5134)

## Description

This change will change from accepting any rows that have `openshift-project`, `openshift-namespace` or `openshift-cluster` as a key to look for cluster specific tag key value pairs. Using just the key caused false positive resource matching when dealing with multiple clusters connected to the same cloud source containing the `openshift-project` as a key. 

Therefore, I needed to update the code to look for possible key value pairs instead. This update means the only data stored in the managed data table is correlated data. Therefore, I needed to update our validation logic to address this new hurdle. I could not rely on the matched_tag column in the parquet table because it was not getting populated with the keys when an `openshift-` was matched. Therefore, I used the logic to find the expected values to solve this problem. 

## Testing

```
ENV_FOR_DYNACONF=local DYNACONF_IQE_VAULT_LOADER_ENABLED=false DYNACONF_IQE_VAULT_IGNORE_PATH_ERRORS=True iqe tests plugin cost_management -k test_api_ocp_on_gcp_source_raw_calc --pdb

trino:org1234567> select sum(cost) from managed_gcp_openshift_daily;
  _col0
----------
 562080.0
(1 row)

Query 20240918_201908_01265_4fsir, FINISHED, 1 node
http://localhost:8080/ui/query.html?20240918_201908_01265_4fsir
Splits: 107 total, 107 done (100.00%)
CPU Time: 0.1s total, 35.9K rows/s, 8.96MB/s, 7% active
Per Node: 0.1 parallelism, 4.18K rows/s, 1.04MB/s
Parallelism: 0.1
Peak Memory: 1.5KB
0.60 [2.51K rows, 642KB] [4.18K rows/s, 1.04MB/s]

trino:org1234567> select sum(unblended_cost) from reporting_ocpgcpcostlineitem_project_daily_summary where gcp_source='6057a987-5bae-4e64-975f-59aafa8902a9';
  _col0
----------
 562080.0
(1 row)
```

Run the smokes on main and this branch and notice how the cost don't match on main, but match on this branch.

## Release Notes
- [ ] proposed release note

```markdown
* [COST-5134](https://issues.redhat.com/browse/COST-5134) Improve tag matching logic
```

### CODE REVIEW

File: koku/masu/database/trino_sql/verify/managed_ocp_on_aws_verification.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    WHERE managed_ocpcloud.source = {{cloud_source_uuid}}
+    AND managed_ocpcloud.year = {{year}}
+    AND managed_ocpcloud.month = {{month}}
+    AND (resource_id_matched = True or matched_tag != '')
+    AND lineitem_lineitemtype != 'SavingsPlanCoveredUsage'
+
+) t1,
+(
+    SELECT sum(unblended_cost) as parquet_total_cost
+    FROM hive.org1234567.reporting_ocpawscostlineitem_project_daily_summary as parquet_table
```
Comment (lcouzens) [inline koku/masu/database/trino_sql/verify/managed_ocp_on_aws_verification.sql:19]: Pretty sure this is incorrect but I could be wrong :grin:

Comment (myersCody) [reply koku/masu/database/trino_sql/verify/managed_ocp_on_aws_verification.sql:19]: lol it is still a draft but I appreciate the call out

**Comment by myersCody** [comment]
After the change I needed to rework the verification logic:
*GCP*
```
koku-worker-1  | [2024-10-02 16:25:15,069] INFO aca09a0d-8258-4b44-acfc-5950618452b5 718 {'message': 'Verification successful', 'tracing_id': '', 'schema': 'org1234567', 'cloud_source_uuid': '910aadc9-59a6-449a-8eb9-a3fd347a8233', 'year': '2024', 'month': '09', 'matched_tag_array': ['"openshift_cluster": "test_cost_ocp_on_gcp_cluster_kikis"', '"openshift_node": "test_gcp_kikis"', '"openshift_project": "test_kikis"', '"openshift_project": "test_pom"', '"openshift_node": "test_gcp_pom"', '"openshift_cluster": "test_cost_ocp_on_gcp_cluster_pom"']}
```

*AWS*
```
koku-worker-1  | [2024-10-02 16:58:55,421] INFO b1180379-2fe4-4871-bff2-373f93b576ca 43 {'message': 'Verification successful', 'tracing_id': '', 'schema': 'org1234567', 'cloud_source_uuid': 'b76e7e68-6fc9-446d-88de-17939addbaa9', 'year': '2024', 'month': '09', 'matched_tag_array': ['"openshift_node": "ocp-nod"', '"stack": "overflow"', '"openshift_cluster": "test_cost_ocp_on_aws_cluster"', '"volume": "stor_nod"', '"openshift_project": "openshift-ocp_prod"', '"app": "nod"', '"git": "commit"']}
```

*Azure*
```
koku-worker-1  | [2024-10-02 17:02:55,602] INFO b4a16081-3cc5-459e-98ee-b956044ea4cd 43 {'message': 'Verification successful', 'tracing_id': '', 'schema': 'org1234567', 'cloud_source_uuid': '040fa926-aaa5-456c-bbca-a37920ed3e64', 'year': '2024', 'month': '09', 'matched_tag_array': ['"openshift_cluster": "test_cost_ocp_on_azure_cluster"', '"openshift_project": "test_chilli"', '"openshift_project": "test_victor"', '"openshift_node": "test_azure_victor"']}
```

**Review by lcouzens — APPROVED** [review]
Great job Cody! This is looking sweet.

**Review by lcouzens — APPROVED** [review]

**Pull request merged**